library/configuration/ftrack/library/configuration/configuration.py

```diff
@@ -0,0 +1,471 @@
+# :coding: utf-8
+# :copyright: Copyright (c) 2024 ftrack
+
+"""
+This module provides functionality to load, merge, resolve and save configuration files.
+The utilities in this module are designed to work nicely together.
+They are supposed to be run in sequence. For example:
+
+>>> register_resolvers()
+>>> config_files = get_configurations_from_entrypoint("connect.configuration")
+>>> more_config_files = get_configurations_from_namespace("ftrack.library")
+>>> all_config_files = config_files.union(more_config_files)
+
+>>> composed_configuration = compose_configuration_from_configurations(config_files)
+>>> resolved_configuration = resolve_configuration(composed_configuration)
+>>> save_configuration_to_yaml(resolved_configuration, Path("resolved.yaml"))
+"""
+
+import glob
+import importlib
+import itertools
+import logging
+import os
+import pkgutil
+import platform
+
+import platformdirs
+import re
+import socket
+import time
+
+from copy import deepcopy
+from importlib.metadata import entry_points
+from pathlib import Path
+from types import ModuleType
+from typing import (
+    Any,
+    Union,
+)
+
+from omegaconf import OmegaConf, DictConfig, ListConfig
+from pydantic.v1.utils import deep_update
+
+# log = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+METADATA_ROOT_KEY = "_metadata"
+METADATA_SOURCES_KEY = "sources"
+METADATA_DELETE_KEY = "delete-after-compose"
+METADATA_CONFLICTS_KEY = "conflicts"
+
+
+class Configuration:
+    def __init__(
+        self,
+        package_name: str = "",
+        file_path: Path = "",
+        configuration: DictConfig = OmegaConf.create({}),
+    ):
+        self.package_name = package_name
+        self.file_path = file_path
+        self.configuration = configuration
+
+    # TODO: This should probably just be the string representation
+    @property
+    def identifier(self):
+        if self.package_name and self.file_path:
+            return f"{self.package_name}:{self.file_path.name}"
+        elif self.file_path:
+            return self.file_path
+        else:
+            return None
+
+    def __hash__(self):
+        return hash(self.file_path)
+
+
+def register_resolvers() -> None:
```

Comment:
rename to register_default_resolvers and save the register_resolvers to create a wrapper to call the omegaconf like:
```
def register_resolver(id, func, cache):
     OmegaConf.register_new_resolver(
        id,func, use_cache=cache
    )
```